controllers/capabilities/application_plan_reconciler.go

```diff
@@ -111,6 +111,18 @@ func (a *applicationPlanReconciler) syncPlan(_ interface{}) error {
 		}
 	}
 
+	planEntityStateIsPublished := a.planEntity.State() == "published" // If the state is not published then we assume it is "hidden"
+	desiredApplicationPlanIsPublished := a.resource.Published != nil && *a.resource.Published
```

Comment:
minor thing: better to implement `Published()` method in the `capabilitiesv1beta1.ApplicationPlanSpec`

Reply:
:+1: created

Reply:
I do not see it

Reply:
you should see it now